helix-core/src/object.rs

```diff
@@ -40,6 +41,85 @@ pub fn select_next_sibling(syntax: &Syntax, text: RopeSlice, selection: Selectio
     })
 }
 
+fn find_parent_with_more_children(mut node: Node) -> Option<Node> {
```

Comment:
I think thus also needs to be able to cross injection layer so I would use the new tree cursor here